ql/src/java/org/apache/hadoop/hive/ql/parse/SemanticAnalyzer.java

```diff
@@ -12550,6 +12550,21 @@ private ParseResult rewriteASTWithMaskAndFilter(TableMask tableMask, ASTNode ast
     }
   }
 
+  void gatherUserSuppliedFunctions(ASTNode ast) {
+    int tokenType = ast.getToken().getType();
+    if (tokenType == HiveParser.TOK_FUNCTION ||
+            tokenType == HiveParser.TOK_FUNCTIONDI ||
+            tokenType == HiveParser.TOK_FUNCTIONSTAR) {
+      if (ast.getChild(0).getType() == HiveParser.Identifier) {
+        // maybe user supplied
+        this.userSuppliedFunctions.add(ast.getChild(0).getText());
```

Comment:
I'm not sure of Hive's stance on this - but I usually only use this when forced to. Like in a setter in which the in parameter is named the same as the object variable. Feel free to ignore if you disagree.

Reply:
I think this is the way to set field variables into the semantic analyzer object. Only then I can retrieve it in the `CommandAuthorizerV2` class

Reply:
The `ast.getChild(0).getText()` should be trimmed by `unescapeIdentifier(expressionTree.getChild(0).getText())`.

Reply:
Ack